src/app/services/error-handler/error-interceptor.service.ts

```diff
@@ -0,0 +1,44 @@
+import { HttpEvent, HttpHandler, HttpInterceptor, HttpRequest } from '@angular/common/http';
+import { Injectable } from '@angular/core';
+import { MatSnackBar } from '@angular/material/snack-bar';
+import { catchError, Observable } from 'rxjs';
+import { RegistryError } from 'src/app/directives';
+
+@Injectable()
+export class ErrorInterceptorService implements HttpInterceptor {
+	constructor(private matSnackBar: MatSnackBar) {}
```

Comment:
Could we follow the pattern of using `private readonly` for our injectables? It should be common knowledge to not change the reference to an injected service, but I like the idea of marking which things should only be read and can't be overwritten